modules/services/window-managers/i3-sway/sway.nix

```diff
@@ -0,0 +1,311 @@
+    moduleName = "sway";
+  };
+
+  inherit (commonFunctions) keybindingsStr keycodebindingsStr modeStr assignStr barStr gapsStr floatingCriteriaStr windowCommandsStr colorSetStr;
+
+  startupEntryStr = { command, always, ... }: ''
+    ${if always then "exec_always" else "exec"} ${command}
+  '';
+
+  inputStr = name: attrs: ''
+    input "${name}" {
+    ${concatStringsSep "\n" (mapAttrsToList (name: value: "${name} ${value}") attrs)}
+    }
+  '';
+
+  outputStr = name: attrs: ''
+    output "${name}" {
+    ${concatStringsSep "\n" (mapAttrsToList (name: value: "${name} ${value}") attrs)}
+    }
+  '';
+
+  configFile = pkgs.writeText "sway.conf" ((if cfg.config != null then with cfg.config; ''
+    font pango:${concatStringsSep ", " fonts}
+    floating_modifier ${floating.modifier}
+    default_border ${if window.titlebar then "normal" else "pixel"} ${toString window.border}
+    default_floating_border ${if floating.titlebar then "normal" else "pixel"} ${toString floating.border}
+    hide_edge_borders ${window.hideEdgeBorders}
+    focus_wrapping ${if focus.forceWrapping then "yes" else "no"}
+    focus_follows_mouse ${if focus.followMouse then "yes" else "no"}
+    focus_on_window_activation ${focus.newWindow}
+    mouse_warping ${if focus.mouseWarping then "output" else "none"}
+    workspace_layout ${workspaceLayout}
+    workspace_auto_back_and_forth ${if workspaceAutoBackAndForth then "yes" else "no"}
+
+    client.focused ${colorSetStr colors.focused}
+    client.focused_inactive ${colorSetStr colors.focusedInactive}
+    client.unfocused ${colorSetStr colors.unfocused}
+    client.urgent ${colorSetStr colors.urgent}
+    client.placeholder ${colorSetStr colors.placeholder}
+    client.background ${colors.background}
+
+    ${keybindingsStr keybindings}
+    ${keycodebindingsStr keycodebindings}
+    ${concatStringsSep "\n" (mapAttrsToList inputStr input)}
+    ${concatStringsSep "\n" (mapAttrsToList outputStr output)}
+    ${concatStringsSep "\n" (mapAttrsToList modeStr modes)}
+    ${concatStringsSep "\n" (mapAttrsToList assignStr assigns)}
+    ${concatStringsSep "\n" (map barStr bars)}
+    ${optionalString (gaps != null) gapsStr}
+    ${concatStringsSep "\n" (map floatingCriteriaStr floating.criteria)}
+    ${concatStringsSep "\n" (map windowCommandsStr window.commands)}
+    ${concatStringsSep "\n" (map startupEntryStr startup)}
+  '' else "") + "\n" + (if cfg.systemdIntegration then ''
+    exec "systemctl --user import-environment; systemctl --user start sway-session.target"
+  '' else "") + cfg.extraConfig);
+in
+
+{
+  options.wayland.windowManager.sway = {
+    enable = mkEnableOption "sway wayland compositor";
+
+    package = mkOption {
+      type = with types; nullOr package;
+      default = pkgs.sway;
+      defaultText = literalExample "${pkgs.sway}";
+      description = ''
+        Sway package to use. Set to null to not install a package (for example if using the nixos option to install sway)
```

Comment:
I think that unnecessarily overcomplicates the matter. I believe people should just pick the package where the module they're using resides, not mix and match between home-manager and nixos.

Reply:
I've looked over this and it should be an easy change to move all the nixos features here